modules/administration/pages/auth-methods-sso.adoc

```diff
@@ -0,0 +1,209 @@
++
+----
+docker run -td --name=idp -p 8080:8080 -e KEYCLOAK_USER=<user> -e KEYCLOAK_PASSWORD=<password> quay.io/keycloak/keycloak:9.0.2
+----
+. Sign in as a privileged user, and use the Keycloak {webui} to create a realm using these details:
++
+* In the ``Name`` field, enter a name for the realm.
+For example, ``SUMA``.
+* Toggle the ``Enabled`` switch to ``On``.
+* In the ``Endpoints`` field, type ``SAML 2.0 Identity Provider Metadata``.
+This endpoint is ``<IdP_Address>/auth/realms/<Realm_Name>/protocol/saml/descriptor`` which describes the endpoints and certificate in the {productname} configuration file.
+// Probably needs more explanation here. --LKB 20200415
+
+
+
+When you have Keycloak running and set up, you can add the endpoints.
+Keycloak refers to endpoints as clients.
+
+
+
+.Procedure: Adding Endpoints as Clients
+. In the Keycloak {webui}, create a new client using these details:
++
+* In the ``Client ID`` field, enter the endpoint specified in the server configuration file as ``java.sso.onelogin.saml2.idp.entityid``.
+For example, ``https://<FQDN>/rhn/manager/sso/metadata``.
+* In the ``Client Protocol`` field, select ``SAML``.
+* Toggle the ``Includew AuthnStatement`` switch to ``On``.
+* Toggle the ``Sign Assertions`` switch to ``On``.
+* In the ``Signature Algorithm`` field, select ``RSA_SHA1``.
+* In the ``SAML Signature Key Name`` field, select the key.
+* In the ``Canonicalization Method`` field, select ``Exclusive``.
+. In the ``Fine Grain SAML Endpoint Configuration`` section, add the two endpoints using these details:
++
+* In both the ``Assertion Consumer Service`` fields, enter the endpoint specified in the server configuration file as ``java.sso.onelogin.saml2.sp.assertion_consumer_service.url``.
+For example, ``https://<FQDN>/rhn/manager/sso/acs``.
+* In both the ``Logout Service`` fields, enter the endpoint specified in the server configuration file as ``java.sso.onelogin.saml2.sp.single_logout_service.url``.
+For example, ``https://<FQDN>/rhn/manager/sso/sls``.
+
+
+
+When you have added the endpoints as clients, you can configure the client scope, and map the users between Keycloak and {productname}.
+
+
+
+.Procedure: Configuring Client Scope and Mappers
+. In the Keycloak {webui}, navigate to the menu:Clients[Client Scopes] tab and assign ``role_list`` as the default client scope.
+. Navigate to the menu:Clients[Mappers] tab and add a user attribute ``Uid`` mapper, using the default values.
+This SAML attribute is expected by {productname}.
+. Navigate to the menu:Users[Admin] section and create an administrative user.
+This user does not need to match the {productname} administrative user.
+. Navigate to the menu:Users[Role Mappings] tab, add a ``uid`` attribute with a value that matches the username of the {productname} administrative user.
+. Navigate to the menu:Users[Credentials] tab, and set the same password as used by the {productname} administrative user.
```

Comment:
strange. above we say that this admin user is different from the productname admin user.  Why using the same password?

Reply:
That's probably a @srbarrios question. My understanding was that you create a user (that can be basically anything), and then give it the appropriate parameters so it authenticates correctly to SUMA.

Reply:
Yes, you are right @keichwa, my bad, so the point there is that when I did the integration with Keycloak I already had setup a user in SUMA, so on my understanding, to not confuse the customer, we use the same password here.
But in fact, you can use a different password, the only problem is that the customer must know that needs to use that new password. So.. the passwords added on SUMA are not anymore usable.

Reply:
Probably simpler in this case to just say to use the same here. If they're using LDAP it's a different procedure anyway.